Feature/logout

ASHmd97 opened the pull request:
## Pull Request Description

### Title: Add Logout Feature with Dialog

### Description:
This PR adds the logout feature to the app. It includes a confirmation dialog, navigation to the login screen on success, and an error message if logout fails.

### What I Did:
- **Added Logout Dialog in `LogoutDialog`:**
  - Created a dialog with "Cancel" and "Log Out" buttons to confirm logout.
  - Used `BlocConsumer` to listen to `ProfileCubit` states (`logoutState`):
    - Shows a `CircularProgressIndicator` during `BaseLoadingState`.
    - On success (`BaseSuccessState`), navigates to the login screen using `Navigator.pushReplacementNamed(context, Routes.login)`.
    - On error (`BaseErrorState`), closes the dialog and shows a black `SnackBar` with the error message in white text using `ScaffoldMessenger`.
  - Passed `ProfileCubit` as a parameter to `LogoutDialog` to avoid context issues.

- **Updated `ProfilePage` to Support Logout:**
  - Added `final cubit = context.read<ProfileCubit>();` to pass the cubit to `LogoutDialog`.
  - Added a comment to explain why:
    ```dart
    // Getting ProfileCubit here to send it to LogoutDialog, 
    // because Dialog can't see ProfileCubit on its own.
    ```

- **Added Logout Logic in `ProfileCubit`:**
  - Implemented `logout()` method to call `LogoutUseCase`.
  - Updated `ProfileState` to include `logoutState` for tracking loading, success, and error states.

- **Integrated Backend for Logout:**
  - Created `LogoutUseCase` to handle logout logic.
  - In `ProfileRepositoryImpl`:
    - Called the logout API via `ProfileApiRemoteDataSource` using the endpoint `ApiConstants.logOutEndPoint`.
    - Deleted the token from local storage using `ProfileLocalDataSourceImpl` by calling `deleteToken()`.

- **Added Local Storage Handling:**
  - Implemented `ProfileLocalDataSourceImpl` to delete the token using `LocalStorageClient`:
    ```dart
    await _localStorageClient.deleteData('token');
    ```

- **Added Remote API Call for Logout:**
  - In `ProfileApiRemoteDataSource`, implemented `logout()` to call the logout endpoint with a token:
    ```dart
    final response = await _apiClient.get(
      ApiConstants.logOutEndPoint,
      requiresToken: true,
    );
    ```

### Why I Did This?
- To let users log out safely with a confirmation dialog.
- To show feedback (loading, success, or error) so the user knows what's happening.
- To make sure the token is deleted and the API is called when logging out.

### Testing:
- Tested clicking "Log Out":
  - Shows a loading spinner for 5 seconds (simulated delay).
  - Goes to the login screen if logout works.
  - Shows a black `SnackBar` with the error message if it fails.
- Made sure the dialog closes properly in all cases.

ASHmd97 committed:
Merge branch 'devolpment' into feature/logout

ASHmd97 committed:
feat(logout): implement logout functionality

- Added logout endpoint to ApiConstants.
- Updated ProfileRemoteDataSource and ProfileRepository to include logout method.
- Integrated logout use case in ProfileCubit.
- Enhanced ProfileState to manage logout state.
- Updated UI components to handle logout actions and display appropriate messages.

ASHmd97 committed:
Merge branch 'devolpment' into feature/logout

MoatazAbdElwhab commented on lib/features/profile/presentation/widgets/logout_dialog.dart:
```diff
@@ -0,0 +1,132 @@
+            ),
+            SizedBox(height: 8.h),
+            Text(
+              'Confirm logout!!',
```
don't use hard code string

ASHmd97 replied:
I solved it

MoatazAbdElwhab commented on lib/features/profile/presentation/cubit/profile_cubit.dart:
```diff
@@ -93,6 +96,20 @@ class ProfileCubit extends Cubit<ProfileState> {
 
+  Future<void> logout() async {
+    emit(state.copyWith(logoutState: BaseLoadingState()));
+    await Future.delayed(Duration(seconds: 5));
```
why use Future.delayed ?

ASHmd97 replied:
to simulate loading state
 i will remove it

MoatazAbdElwhab reviewed (commented):
good work

shimaahossni commented on lib/features/profile/presentation/widgets/logout_dialog.dart:
```diff
@@ -0,0 +1,132 @@
+                        if (state.logoutState is BaseErrorState) {
+                          Navigator.pop(context);
+                          ScaffoldMessenger.of(context).showSnackBar(
+                            SnackBar(
```
dont use ScaffoldMessenger.of(context).showSnackBar....try to use  GetIt.I<DialogUtils>().showSnackBar ....

ASHmd97 replied:
tmam

ASHmd97 committed:
fix(logout): enhance logout dialog

- Added localized strings for logout functionality in Arabic and English.
- Updated LogoutDialog to use localized strings for titles and messages.
- Removed unnecessary delay in logout process in ProfileCubit.
- Improved error handling in LogoutDialog with a snackbar utility.

ASHmd97 committed:
Merge branch 'feature/logout' of https://github.com/MoatazAbdElwhab/flower_app into feature/logout

MoatazAbdElwhab committed:
Merge branch 'devolpment' into feature/logout
